coala_quickstart/generation/FileGlobs.py

```diff
@@ -0,0 +1,136 @@
+    :return:
+        The list of glob expressions.
+    """
+    response = ask_question(
+        printer,
+        prompt,
+        default=default)
+
+    return comma_split(response)
+
+
+def get_project_files(log_printer, printer, project_dir):
+    """
+    Gets the list of files matching files in the user's project directory
+    after prompting for glob expressions
+
+    :param log_printer:
+        A ``LogPrinter`` object.
+    :param printer:
+        A ``ConsolePrinter`` object.
+    :return:
+        A list of file paths matching the files.
+    """
+    printer.print(glob_help)
+    file_globs = ask_glob_list(
+        printer,
+        "Which files do you want coala to run on?")
+
+    ignore_globs = None
+    if os.path.isfile(os.path.join(project_dir, ".gitignore")):
+        gitignore_globs = get_gitignore_glob(project_dir)
+        gitignore_globs.append(os.path.join(project_dir, ".git/**"))
+        response = ask_question(
+            printer,
+            "Ignore files matching patters in `.gitignore` ?",
+            default="yes")
+        if response.lower() in TRUE_STRINGS:
+            ignore_globs = gitignore_globs
+    if ignore_globs == None:
+        ignore_globs = ask_glob_list(
+            printer,
+            "Which files do you want coala to ignore?",
+            default="")
+    printer.print()
+
+    file_path_globs = [os.path.join(
+        project_dir, glob_exp) for glob_exp in file_globs]
+    ignore_path_globs = [os.path.join(
+        project_dir, glob_exp) for glob_exp in ignore_globs]
+
+    file_paths = collect_files(
+        file_path_globs,
+        log_printer,
+        ignored_file_paths=ignore_path_globs)
+
+    return file_paths
+
+
+def generate_glob_exps(project_files, extensions, project_dir):
+    """
+    Generate glob expressions that matches exactly the files
+    with the given extensions in the user's project directory.
+
+    :param project_files:
+        The list of files to match.
+    :param extensions:
+        A list of file extensions.
+    :param project_dir:
+        The user's project directory.
+    :return:
+        A tuple with the first element (a list of glob expressions)
+        representing files to include and the second element (a list
+        of glob expressions) representing the files to exclude.
+    """
+    globs = []
+    extlist = set()
+    included_globs = []
+    excluded_globs = []
+
+    for root, dir, files in os.walk(project_dir):
+        included_files = {}
+        excluded_files = {}
+        for file in files:
+            name, ext = os.path.splitext(file)
+            if ext not in extensions:
+                continue
+
+            file = os.path.join(root, file)
+            if file in project_files:
+                included_files.setdefault(ext, []).append(file)
+            else:
+                excluded_files.setdefault(ext, []).append(file)
+
+        for ext in included_files:
```

Comment:
im not sure what ext is. my only guess is extension. the name is not so explicit imo